test/e2e/framework/pods.go

```diff
@@ -122,33 +122,35 @@ func (c *PodClient) DeleteSync(name string, options *api.DeleteOptions, timeout
 
 // mungeSpec apply test-suite specific transformations to the pod spec.
 func (c *PodClient) mungeSpec(pod *api.Pod) {
-	if TestContext.NodeName != "" {
-		Expect(pod.Spec.NodeName).To(Or(BeZero(), Equal(TestContext.NodeName)), "Test misconfigured")
-		pod.Spec.NodeName = TestContext.NodeName
-		// Node e2e does not support the default DNSClusterFirst policy. Set
-		// the policy to DNSDefault, which is configured per node.
-		pod.Spec.DNSPolicy = api.DNSDefault
-
-		if !TestContext.PrepullImages {
-			return
-		}
-		// If prepull is enabled, munge the container spec to make sure the images are not pulled
-		// during the test.
-		for i := range pod.Spec.Containers {
-			c := &pod.Spec.Containers[i]
-			if c.ImagePullPolicy == api.PullAlways {
-				// If the image pull policy is PullAlways, the image doesn't need to be in
-				// the white list or pre-pulled, because the image is expected to be pulled
-				// in the test anyway.
-				continue
-			}
-			// If the image policy is not PullAlways, the image must be in the white list and
-			// pre-pulled.
-			Expect(ImageWhiteList.Has(c.Image)).To(BeTrue(), "Image %q is not in the white list, consider adding it to CommonImageWhiteList in test/e2e/common/util.go or NodeImageWhiteList in test/e2e_node/image_list.go", c.Image)
-			// Do not pull images during the tests because the images in white list should have
-			// been prepulled.
-			c.ImagePullPolicy = api.PullNever
+	if !TestContext.NodeE2E {
+		return
```

Comment:
This logic looks wrong... if the `NodeName` isn't set, but `PrepullImages` is, we should still update the `ImagePullPolicy`.  (I know this is the same as the previous logic, but that looks wrong too)